# *C. difficle* Gut Microbiome Colonization in Relation to Diet Study 

This repository includes all files produced/required to run Qiime2 core diversity analysis on the total sum scaled biom table for the gut microbiome study along with the subsequent created visualizations and statistical analysis performed in R. The jupyter notebook that includes the script for the core diversity analysis in Qiime2 is included. Files needed to run PICRUSt2 for gut microbiome functional potential predictions are included as well as the R scripts for output visualizations and statistical analyses. The workflow directory contains the scripts needed to run all data analysis done so far via Snakemake (please run this on a HPC of some sort - it'll save your computer). 

## To Do: 

  - upload pred_metagenome_contrib.tsv.gz at some point
  - analysis and pairwise comparisons of histopathology, toxin, metabolomics, hypoxia, mouse weight, and bile acid data
  - put together figures and outline for paper manuscript
  - check to see if there are inflammation differences between the two barrier facilities (studies 3 and 7 are from one, and studies 1,2,4,5, and 6 are from the other - encode data based on barrier facitlity)


  
