# The effects of dietary fat and fiber on the gut microbiome in the context of antibiotic-induced *C. difficle* infection

This repository includes all files produced through QIIME2 microbiome profiling of 16S rDNA sequencing FASTQ files along with visualizations and statistical analysis run in R. Analysis was optimized through a pipeline generated in Snakemake, associated files are contained in the workflow directory. 

## Directory contents are as follows:
> [!IMPORTANT] 
> Each directory contains a `README.md` file that includes additional information about the code/data provided.

- **amc_followUp_experiments:** Follow-up experiments conducted at AMC (Anschutz Medical Campus)
- **arizona_experiments:** Main mouse experiments done at the University of Arizona 
- **comparisons:** A combination of baseline (day -15) and day 3 microbiome data from the Arizona experiments, AMC follow-up experiments, and previously published data from [this study](https://www.nature.com/articles/s41522-022-00276-1#ref-CR67) to make comparisons between studies.
- **figures:** Compiled manuscript figures, plots taken from all directories. 
- **workflow:** Contains the pipeline built for the purpose of this data analysis in Snakemake (the [tutorial](https://github.com/madiapgar/diet_mouse_cdiff/tree/master/workflow/tutorial) is here!).


## Links (will be hyperlinked once available):
- paper
- raw 16S sequences 

## R Package: apppleplots
This project has an associated R package named [apppleplots](https://github.com/madiapgar/apppleplots) that contains functions to create the plots/stats seen in the paper (and this repo). All plots featured in the paper are also included in the package as `ggplot2` objects. 

## Contact Information: 
All code and subsequent outputs were generated by Madison Apgar, a graduate student in the Lozupone lab at the University of Colorado Anschutz Medical Campus. 

- **LinkedIn:** Madi Apgar
- **Email:** madison.apgar@cuanschutz.edu


  
