---
title: "manuscript_figures"
output: html_document
date: "2023-10-02"
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

```{r}
library(tidyverse)
library(cowplot)
library(patchwork)
# library(svgparser)
# library(cssparser)
# library(grid)
library(dmcdmfigs)
```

# **Putting Together Figures for Paper Manuscript**
These are, of course, subject to change but I mine as well start somewhere. The order of the figures will probably change as well as I figure out the story I want to tell with this manuscript. 


## **Figure One**
Plots to go in figure:
- experimental timeline diagram (need to redo this)
- mouse survival curve plot and stats

*Caption*:
*a) Survival curves for the five tested diets after antibiotic induced gut microbiome dysbiosis and C. difficile infection. b) A heat map of log-normal pairwise comparisons of survival probability between the tested diets. P-value significance (****p<0.0001, ***p<0.001, **p<0.01, *p<0.05, +p<0.1, ns<1).*

```{r}
# timeline_FP <- '../../figures/test_timeline.svg'
# 
# timeline <- svgparser::read_svg(timeline_FP, obj_type = 'data.frame')
# 
# grid::grid.draw(timeline)
```


```{r, warning=FALSE, fig.height=12, fig.width=15}
fig1_bottom_row <- plot_grid(surv_stat_vis, NULL,
                             rel_widths = c(0.6, 1),
                             labels = c('b)', ' '),
                             label_size = 20)

plot_grid(diet_plot_final +
            theme_bw(base_size = 18),
          fig1_bottom_row,
          rel_heights = c(1, 0.5),
          nrow = 2,
          labels = c('a)', ' '),
          align = 'h',
          label_size = 20) -> fig1

fig1
```

## **Figure Two**
Plots to go in figure:
- toxin plot (neat)
- histopathology plot
- toxin and histopathology comparison plot and stats

*Caption*:
*a) C. difficile TcdA and TcdB toxin concentrations in the five tested diets, determined by ELISA. b) Histologic inflammation scoring of mouse colon and cecal contents at three days post C. difficile infection in the tested diets as determined by a blinded histologist. Statistical analysis for score differences was done using a Kruskal-Wallis test followed by a Dunn's Post Hoc test with p-values being corrected for multiple comparisons via Benjamini and Hochberg. c) Linear regression comparison of histopathology scores to TcdA burden across tested diets. d) Linear regression comparison of histopathology scores to TcdB burden across tested diets. e) Linear regression results for c) and d) as a heat map with the color indicating the correlation coefficient with the overlaid text being the p-value significance* (****p<0.0001, ***p<0.001, **p<0.01, *p<0.05, +p<0.1, ns<1).
```{r, warning=FALSE, fig.height=17, fig.width=26.5}
fig2_top_row <- plot_grid(neat_plot +
                            theme_bw(base_size = 20), 
                          all_day_plot,
                          labels = c('a)', 'b)'),
                          rel_widths = c(0.9, 1),
                          align = 'h',
                          label_size = 20)

fig2_middle_row <- plot_grid(neat_histo_tca_plot +
                               theme_bw(base_size = 20) +
                               theme(strip.text.y.right = element_text(angle = 0)), 
                             neat_histo_tcb_plot +
                               theme_bw(base_size = 20) +
                               theme(strip.text.y.right = element_text(angle = 0)),
                             labels = c('c)', 'd)'),
                             align = 'h',
                             ncol = 1, 
                             label_size = 20)

fig2_stats <- plot_grid(neat_tox_stat_vis +
                               theme_bw(base_size = 20) +
                               ggtitle('Histopathology Score and Toxin\n Concentration Comparisons'),
                        NULL,
                        ncol = 1,
                        rel_heights = c(1, 0.8),
                        labels = c('e)', ' '),
                        label_size = 20)

fig2_bottom_row <- plot_grid(fig2_middle_row,
                             fig2_stats,
                             nrow = 1,
                             rel_widths = c(1, 0.6))

plot_grid(fig2_top_row,
          fig2_bottom_row,
          rel_heights = c(0.8, 1),
          ncol = 1) -> fig2

fig2
```

## **Figure Three**
Plots to go in figure:
- bile acid concentration plots (we don't have these yet)
- hypoxia data? 
```{r}

```

## **Figure Four**
Plots to go in figure:
- metabolomics plot
- metab and histo comparison plot and stats

*Caption*:
*a) SCFA (acetate, butyrate, propionate) levels of mouse cecal contents three days post C. difficile infecion by the tested diets. Linear regression comparison of SCFAs, b) acetate, c) butyrate, and d) propionate to histopathology scores across tested diets. e) Linear regression results for b)-d) as a heat map with the color indicating the correlation coefficient with the overlaid text being the p-value significance* (****p<0.0001, ***p<0.001, **p<0.01, *p<0.05, +p<0.1, ns<1).*
```{r, warning=FALSE, fig.height=20, fig.width=32}
fig4_bottom_row <- plot_grid(histo_acetic_plot +
                               theme_bw(base_size = 20) +
                               theme(strip.text.y.right = element_text(angle = 0)),
                             histo_butanoic_plot +
                               theme_bw(base_size = 20) +
                               theme(strip.text.y.right = element_text(angle = 0)),
                             histo_propanoic_plot +
                               theme_bw(base_size = 20) +
                               theme(strip.text.y.right = element_text(angle = 0)),
                             labels = c('c)', 'd)', 'e)'),
                             align = c('hv'),
                             axis = c('tblr'),
                             nrow = 2,
                             label_size = 20)

fig4_top_row <- plot_grid(metab_plot,
                          metab_stat_vis +
                               theme_bw(base_size = 20)+
                               ggtitle('Histopathology Score and Metabolite\n Concentration Comparisons'),
                          nrow = 1,
                          rel_widths = c(1, 0.6),
                          labels = c('a)', 'b)'),
                          label_size = 20)
plot_grid(fig4_top_row,
          fig4_bottom_row,
          ncol = 1,
          rel_heights = c(0.65, 1)) -> fig4

fig4
```

## **Figure Five**
Plots to go in figure:
- change all to the unweighted unifrac instead of the weighted unifrac!!!

*Caption*:
*a) Unweighted UniFrac PCoA plots delimited by diet (columns) and by days relative to infection (rows). Individual points are colored by their associated Faith's PD score. b) Microbiome resilience determined using pairwise comparisons between day negative eight (seven days post diet switch and five days prior to oral antibiotic administration) and later time points. c) Microbiome homogeneity determined using pairwise Unweighted UniFrac distances within each time point by tested diets. The dashed red line at day negative three indicates the day the mice received oral antibiotics and the purple dashed line at day zero indicates with the mice were infected with C. difficile. Statistical analysis of microbiome d) resiliency and e) homogeneity was done with a Kruskal-Wallis test followed by a Dunn's Post Hoc test with the color gradient indicating the difference in means between the various groups.*P-value significance* (****p<0.0001, ***p<0.001, **p<0.01, *p<0.05, +p<0.1, ns<1).*
```{r, warning=FALSE, fig.width=26.5, fig.height=18}
## use NULL to create a ghost plot to alter your rows
fig5_bottom_row <- plot_grid(uu_resil_plot, uu_homog_plot,
                             uu_resil_stat_vis +
                               theme(legend.position = "None"), 
                             uu_homog_stat_vis,
                             nrow = 2,
                             rel_heights = c(1, 0.7),
                             rel_widths = c(0.90, 1),
                             labels = c('b)', 'c)', 'd)', 'e)'),
                             label_size = 20)


unweighted_by_day <- na.omit(unweighted_by_day)

my_table_theme <- gridExtra::ttheme_default(core=list(bg_params = list(fill = 'white', col=NULL)))
unweighted_top <- unweighted_pcoa + gridExtra::tableGrob(unweighted_by_day,
                                                     rows = NULL, 
                                                     theme = my_table_theme)
                    
fig5_top_row <- plot_grid(unweighted_top,
                          nrow = 1,
                          # rel_widths = c(1, 0.5),
                          labels = c('a)'),
                          label_size = 20,
                          align = 'v')

plot_grid(fig5_top_row,
          fig5_bottom_row,
          nrow = 2,
          rel_heights = c(0.9, 1)) -> fig5

fig5
```

## **Figure Six**
Plots to go in figure:
- faith's pd and stats

*Caption*:
*a) Faith's Phylogenetic Diversity (alpha diversity) analysis by the five tested diets. The dashed red line at day negative three indicates the day the mice received oral antibiotics and the purple dashed line at day zero indicates with the mice were infected with C. difficile. b) Pairwise comparisons between diets as determined by a Kruskal-Wallis test followed by a Dunn's Post Hoc test with the color gradient indicating the difference in means.*P-value significance* (****p<0.0001, ***p<0.001, **p<0.01, *p<0.05, +p<0.1, ns<1).*
```{r, warning=FALSE, fig.height=9, fig.width=15}
plot_grid(faith_plot +
            theme_bw(base_size = 20), 
          faith_stat_vis,
          ncol = 1,
          rel_heights = c(1, 0.7),
          labels = c('a)', 'b)'),
          align = 'h',
          label_size = 20) -> fig6

fig6
```

## **Figure Seven**
Plots to go in figure:
- want to eventually have the plot that goes with 16S for stool and cecal samples 
- need an unbiased differential abundance test? 
- asv level of what things are influenced by diets 
- look for microbes that correlate with inflammation - use an ANCOMBC 
- filter low abundance asvs (anything that's in less than 20% of the samples)
**- have a distance matrix for every single mouse by their taxonomic information via qiime or r (which taxa are showing more related patterns by diet in mice). then, run an adonis test on it.**

*Caption*:
*a) Relative abundances of key gut bacterial families by the five tested diets throughout the experimental timeline. The dashed red line at day negative three indicates the day the mice received oral antibiotics and the purple dashed line at day zero indicates with the mice were infected with C. difficile. b) Pairwise comparisons by days relative to infection (columns) and bacterial family (rows) per diet as determined by a Kruskal-Wallis test followed by a Dunn's Post Hoc test with the color gradient indicating the difference in means.*P-value significance* (****p<0.0001, ***p<0.001, **p<0.01, *p<0.05, +p<0.1, ns<1).*
```{r, warning=FALSE, fig.height=15, fig.width=35}
plot_grid(short_abun1, stat_plot1,
          labels = c('a)', 'b)'),
          label_size = 20,
          rel_widths = c(1, 0.9)) -> fig7

fig7
```

## **Figure Eight**
TAKING THIS FIGURE OUT!!

Just in case we decide to do more with this data, here's the results text I wrote for it:

*SCFA (butyrate) and secondary bile acid coding capacity by bacterial class and mouse diet.*
	We put the 16S rDNA fecal sample results through PICRUSt2 software to measure the potential functional coding capacity of key SCFA and secondary bile acid producing enzymes throughout our experimental timeline to determine the possible difference between coding capacity and which bacterial classes could be contributing to those differences in our tested diets. Butyrate kinase and butyryl-CoA-transferase are enzymes that aid in producing butyrate from dietary fiber through two different pathways. baiH and baiI are enzymes that play a role in producing secondary bile acids from primary bile acids through hydroxylation. 
	Based on our PICRUSt2 results, we saw that Bacilli, Bacteroidia, and Clostridia were the top bacterial classes contributing to the production of butyrate through the use of butyrate kinase with no major differences seen between our tested diets (Figure 8a,e). Alphaproteobacteria, Bacilli, Bacteroidia, and Clostridia were the top bacterial classes contributing to the production of butyrate through the use of butyryl-CoA- transferase with no major differences seen between our tested diets (Figure 8b,e). Hungatella and Lachnospiraceae were top contributors of baiH and Blautia was the top contributor of baiI for the production of secondary bile acids (Figure 8 c,d) (note: secondary bile acid enzyme contributors are measured at the genus level rather than the class level). Mice on the chow diet had higher butyrate kinase and butyryl-CoA-transferase coding capacity at day 3 than mice on high fiber diets after antibiotic administration and CDI (Figure 8e). These results indicate the top bacterial classes that have coding capacity for butyrate coding capacity and the top bacterial genus’s that have coding capacity for secondary bile acids between the tested diets. Mice on the chow diet are the only diet to show higher butyrate coding capacity at day 3 compared to mice on the high fiber diets. Overall, these results, while interesting, do not contribute in explaining the differences in mouse mortality between diets (Figure 1). 


*Caption*:
*All results come from PICRUSt2 analysis based on the fecal sample 16S sequences. The dashed red line at day negative three indicates the day the mice received oral antibiotics and the purple dashed line at day zero indicates with the mice were infected with C. difficile. a) Abundance of butyrate-kinase coding capacity genes by bacterial class and the tested diets across time points. b) Abundance of butyryl-coa-transferase coding capacity genes by bacterial class and the tested diets across time points. Abundance of c) baiH and d) baiI (secondary bile acid producing enzymes) by bacterial species and the tested diets across time points. e) Pairwise comparisons by days relative to infection (columns) and butyrate producing enzyme (rows) per diet as determined by a Kruskal-Wallis test followed by a Dunn's Post Hoc test with the color gradient indicating the difference in means.*P-value significance* (****p<0.0001, ***p<0.001, **p<0.01, *p<0.05, +p<0.1, ns<1).**
```{r, warning=FALSE, fig.height=29, fig.width=35}
fig8_top_row <- plot_grid(butyrate_kinase, butyryl_coa_transferase,
                          buk_plot +
                               theme(legend.position = "None"), 
                             but_plot,
                          labels = c('a)', 'b)', 'c)', 'd)'),
                          label_size = 20,
                          align = 'hv',
                          axis = 'tblr',
                          nrow = 2,
                          rel_heights = c(1, 0.6))

fig8_bottom_row <- plot_grid(baiH, baiI,
                             baiH_plot +
                               ggtitle('baiH Functional Potential Over Time Between Diets'),
                             NULL,
                             labels = c('e)', 'f)', 'g)'),
                             label_size = 20,
                             align = 'hv',
                             axis = 'tblr',
                             nrow = 2,
                             rel_widths = c(1, 0.7),
                             rel_heights = c(1, 0.7))

plot_grid(fig8_top_row,
          fig8_bottom_row,
          ncol = 1,
          align = 'hv',
          axis = 'tblr',
          rel_heights = c(1, 0.7)) -> fig8

fig8
```

**Saving my Outputs**
as a .pdf and not an rdata object (just to clarify)
```{r}
ggsave('../../figures/fig1.pdf',
       plot = fig1,
       width = 15,
       height = 12)
ggsave('../../figures/fig2.pdf',
       plot = fig2,
       width = 25.5,
       height = 15)
ggsave('../../figures/fig4.pdf',
       plot = fig4,
       width = 32,
       height = 20)
ggsave('../../figures/fig5.pdf',
       plot = fig5,
       width = 26.5,
       height = 18)
ggsave('../../figures/fig6.pdf',
       plot = fig6,
       width = 15,
       height = 9)
ggsave('../../figures/fig7.pdf',
       plot = fig7,
       width = 35,
       height = 15)
# ggsave('../../figures/fig8.pdf',
#        plot = fig8,
#        width = 35,
#        height = 29)
```

