---
title: "manuscript_figures"
output: html_document
date: "2023-10-02"
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

```{r}
library(tidyverse)
library(viridis)
library(ggplot2)
library(cowplot)
library(apppleplots)
```

**Pulling in plots made by keith and laurie (not added to r package yet)**
```{r}
load("immune_cytokine_plot.RData")
load("CDD_cyto_BUN_pcoa.rdat")
load("hflf_butyTox_mediation.rdat")
load("miniComparison_diet_genusAbun_plot.rdat") ## im too lazy to rebuild my r package rn 
load("mini_dietComp_genusAbun_stat_vis.rdat")
load("cfuQuant_dietComp_plot.rdat")
load("all_scfa_comp_plot.rdat")
load("metab.rdata")
load("alphaDiv_microbeProd_plot.rdat")
load("relAbun_microbeProd_plot.rdat")
load("neat_toxin.rdata")
load("noDiet_histoCecum_tox_plot.rdat")
load("histo.rdata")
load("panelC_fig3_redo.rdat")
load("supp_fig2a.rdat")
load("cdiffComparison_genusAbun_plot.rdat")
```

# **Putting Together Figures for Paper Manuscript**


## **Figure 1**
Plots to go in figure:
- experimental timeline diagram 
- mouse survival curve plot and stats

**manuscript specific plot edits**
```{r}
surv_stat_vis_v2 <- surv_stat_vis +
                       scale_fill_viridis(option = 'C',
                                          limits = c(0, 0.05),
                                          direction = 1,
                                          name = 'p-value') +
                        theme(legend.position = 'right')

diet_plot_final_v2 <- diet_plot_final +
                        theme_bw(base_size = 18)

az_spore_weights_plot_v2 <- az_spore_weights_plot +
  theme_bw(base_size = 18) +
  labs(x = 'Days Post Infection',
       y = 'Average % Weight Change',
       title = 'Weight Change After Infection by Diet')
```

**actual manuscript figure**
```{r, warning=FALSE, fig.height=15, fig.width=15}
fig1_middle_row <- plot_grid(surv_stat_vis_v2, NULL,
                             rel_widths = c(0.8, 1),
                             labels = c('d)', ' '),
                             label_size = 20)

plot_grid(diet_plot_final_v2,
          fig1_middle_row,
          az_spore_weights_plot_v2,
          rel_heights = c(1, 0.5, 0.7),
          nrow = 3,
          labels = c('c)', ' ', 'e)'),
          align = 'h',
          label_size = 20) +
  theme(plot.background = element_rect(fill = "white", color = "white")) -> fig1

fig1
```

## **Figure 2**
Plots to go in figure:
- cecal toxin plot 
- cecal overall histopathology plot
- cecal toxin/histo correlation plot (p-value significance should be overlaid) - TAKE OUT!!

**manuscript specific plot edits**
changing the size of the labels on the plot
```{r}
noDiet_histoCecum_toxStats_plot$layers[[4]]$aes_params$size <- 5.5
```

```{r}
neat_plot_v2 <- neat_plot +
                  theme_bw(base_size = 20) +
                  ylab('Concentration (ng/uL)') +
                  ggtitle('Cecum C. difficile Toxin Concentration')

all_day_plot_v2 <- all_day_plot +
                     theme_bw(base_size = 20) +
                     ggtitle('Histopathology Scores') +
                     ylab('Score')

noDiet_histoCecum_toxStats_plot_v2 <- noDiet_histoCecum_toxStats_plot +
                                        theme_bw(base_size = 20)
```


**actual manuscript figure**
```{r, warning=FALSE, fig.height=11, fig.width=20}
fig2_top_row <- plot_grid(neat_plot_v2,
                          all_day_plot_v2,
                             labels = c('a)', 'b)'),
                             align = c('hv'),
                             axis = c('tblr'),
                             nrow = 1,
                            rel_widths = c(0.9, 1),
                             label_size = 20)

fig2_bottom_row <- plot_grid(noDiet_histoCecum_toxStats_plot_v2,
                            labels = c('c)'),
                            label_size = 20)
plot_grid(fig2_top_row,
          fig2_bottom_row,
          ncol = 1,
          align = c('hv'),
          axis = c('tblr'),
          rel_heights = c(1, 0.7)) -> fig2

fig2
```

## **Figure 3**
Plots to go in figure:
- bile acid concentration plot
- bile acid c. diff promoters/inhibitors ratio plot
- bile acid/toxin correlation comparison plots 
- metabolomics plot

**manuscript specific plot edits**
```{r}
dca_sep_plot_v2 <- dca_sep_plot +
                            theme_bw(base_size = 20)

ratio_bile_plot_v2 <- ratio_bile_plot +
                            theme_bw(base_size = 20)+
                            ggtitle('Cecum Bile Acid Ratio')

butyrateOnly_plot_v2 <- butyrateOnly_plot +
                               theme_bw(base_size = 20) +
                               ggtitle('Cecum Butyrate Concentration')
```

**actual manuscript figure**
```{r, fig.height=23, fig.width=24}
fig3_top_row <- plot_grid(dca_sep_plot_v2, 
                          ratio_bile_plot_v2,
                          labels = c('a)', 'b)'),
                          label_size = 20,
                          align = c('hv'),
                          axis = c('tblr'),
                          rel_widths = c(1, 0.5))

fig3_middle_row <- plot_grid(panelC_fig3_redoWithlab,
                             ncol = 1,
                             labels = c('c)'),
                             label_size = 20,
                             align = c('hv'),
                             axis = c('tblr'))


fig3_bottom_row <- plot_grid(metab_plot,
                             all_metab_plotsWithlab,
                             rel_widths = c(0.3, 1),
                             nrow = 1,
                             labels = c('d)', 'e)'),
                             label_size = 20)

(fig3 <- plot_grid(fig3_top_row,
                  fig3_middle_row,
                  fig3_bottom_row,
                  ncol = 1,
                  rel_heights = c(0.5, 0.5, 0.9),
                  align = c('hv'),
                  axis = c('tblr')) +
          theme(plot.background = element_rect(fill = "white", color = "white")))
```

## **Figure 4**
NOTE: stats have been redone log10 transformed

Plots to go in figure:
- faith's pd and stats

```{r}
viridis_pal(option = 'C')(3)
```

**manuscript specific plot edits**
how you change the size of the labels in a ggplot object!!
```{r}
faith_combStats_plot$layers[[4]]$aes_params$size <- 5
```

```{r}
faith_plot_v2 <- faith_plot +
                    theme_bw(base_size = 20)

faith_stat_vis_v2 <- faith_stat_vis +
                        scale_fill_gradient2(low = "#0D0887FF",
                                             high = "#CC4678FF",
                                             name = 'Group 1 -\nGroup 2')
```

**actual manuscript figure**
```{r, warning=FALSE, fig.height=14, fig.width=15}
plot_grid(faith_plot_v2, 
          faith_stat_vis_v2,
          faith_combStats_plot,
          ncol = 1,
          rel_heights = c(1, 0.7, 1),
          labels = c('a)', 'b)', 'c)'),
          align = 'h',
          label_size = 20) -> fig4

fig4
```

## **Figure 5**
NOTE: stats have already been redone log10 transformed!!

Plots to go in figure:
- want to eventually have the plot that goes with 16S for stool and cecal samples 
- need an unbiased differential abundance test? 
- asv level of what things are influenced by diets 
- look for microbes that correlate with inflammation - use an ANCOMBC 
- filter low abundance asvs (anything that's in less than 20% of the samples)
**- have a distance matrix for every single mouse by their taxonomic information via qiime or r (which taxa are showing more related patterns by diet in mice). then, run an adonis test on it.**

```{r}
microbeProd_relAbunStats_plot$layers[[6]]$aes_params$size <- 4.5

microbeProd_relAbunStats_plot_v2 <- microbeProd_relAbunStats_plot +
  theme(legend.position = "top")
```

```{r, warning=FALSE, fig.height=23, fig.width=20}
plot_grid(genusAbun_plots_together_withLabs,
          microbeProd_relAbunStats_plot_v2,
          labels = c(' ', 'c)'),
          ncol = 1,
          label_size = 20,
          rel_heights = c(1, 0.5)) -> fig5

fig5
```

## **Figure 6**
New experiment culture/mortality results.

- include both new plots in this figure!!
keith's plot:
- take color off plot (don't need it to distinguish diets)
- fix x-axis labels
- make point size bigger
- take out the two cytokines: IL12p70, IL-4, IFN-y, IL-2, IL-5

laurie's plot:
- dont need ellipses 
- dont need to worry ab including vendor info 
- make labels bold and bigger
- change transparency of points slightly 
- change color palette to match other figures 
- make lines longer on plot so labels aren't all on top of each other 

figure specific plot changes! (doing them all here so that it get's less confusing)
```{r}
## new fig6a
uu_biplot_v2 <- uu_biplot +
                  theme(legend.position = "none",
                        plot.title = element_blank()) +
                  labs(subtitle = 'PC1-PC2')

uu_biplot2_3_v2 <- uu_biplot2_3 +
                      theme(plot.title = element_blank()) +
                      labs(subtitle = 'PC2-PC3',
                           fill = 'Mouse Diet')

## new fig6b
status_labs_v2 <- c('end_time_course' = 'Survived to End of Time Course',
                    'sick' = 'Humane Endpoint')
location_labs <- c('blood' = 'Blood',
                   'liver' = 'Liver',
                   'spleen' = 'Spleen')

facet_status_cfu_plot_v2 <- facet_status_cfu_plot +
                                guides(fill = guide_legend(override.aes = list(alpha = 1,
                                                                               shape = 21)),
                                       shape = guide_legend(override.aes = list(fill = "black"),
                                                            ncol = 1)) +
                                facet_grid(location~sack_status_f,
                                           labeller = labeller(.cols = status_labs_v2,
                                                               .rows = location_labs)) +
                                labs(title = 'Number of CFUs Cultured from Multiple Locations')

## new fig6c
miniDiet_genusAbun_plot_v2 <- miniDiet_genusAbun_plot +
  labs(subtitle = 'All Experiments')
```


```{r, fig.width=27, fig.height=22}
new_fig6_title <- ggdraw() +
  draw_label('AMC Follow-Up Experiment: Unweighted UniFrac Taxa Biplots',
             size = 24,
             hjust = 1.35)

new_fig6_bottom_row <- plot_grid(uu_biplot_v2, 
                                 uu_biplot2_3_v2,
                                 nrow = 1,
                                 rel_widths = c(0.85, 1))


new_fig6_bottom_row_wLabs <- plot_grid(new_fig6_title,
                                        new_fig6_bottom_row,
                                        nrow = 2,
                                        align = c('hv'),
                                        axis = c('tblr'),
                                        rel_heights = c(0.07, 1),
                                        labels = c('a)'),
                                        label_size = 22)


new_fig6_top_row <- plot_grid(facet_status_cfu_plot_v2,
                              miniDiet_genusAbun_plot_v2,
                              nrow = 1,
                              labels = c('b)', 'c)'),
                              label_size = 22,
                              align = c('v'),
                              axis = c('tb'),
                              rel_widths = c(1, 0.95))

(new_fig6 <- plot_grid(new_fig6_bottom_row_wLabs,
                      new_fig6_top_row,
                      nrow = 2,
                      rel_heights = c(0.75, 1)) +
             theme(plot.background = element_rect(fill = "white", color = "white")))
```


## **Figure 7**
now the plasma immune/sepsis marker data

figure specific plot changes
```{r}
## new fig7a
pcoa_status_labs <- c('end_time_course' = 'Survived to End\nof Time Course',
                      'sick' = 'Humane Endpoint')
p_pcoa_1_2_v2 <- p_pcoa_1_2 +
                  scale_shape_manual(values = c(21,24),
                                     breaks = c('sick',
                                                'end_time_course'),
                                     labels = pcoa_status_labs) +
                  theme(plot.subtitle = element_blank(),
                        legend.position = "right") +
                  labs(fill = 'Mouse Diet')
```

new fig7!
```{r, fig.width=15, fig.height=15}
(fig7 <- plot_grid(p_pcoa_1_2_v2,
                  cytokine_plot,
                  nrow = 2,
                  labels = c('a)', 'b)'),
                  label_size = 22,
                  rel_heights = c(0.9, 1)))
```



## **Supplemental Figure 2**
Plots to go in figure:
- bile acid concentration/CD toxin/histopathology score correlation comparisons
- raw concentrations of acetate and propionate at day 3
- acetate/propionate concentration/CD toxin/histopathology score correlation comparisons

```{r, fig.width=19, fig.height=17}
(figS2_bottom_row <- plot_grid(hflf_buty_plot +
                              labs(title = 'Effects on High-Fat/Low-Fiber Histopathology',
                                   subtitle = 'Butyrate',
                                   x = 'Butyrate (mean-centered)',
                                   y = 'Cecum Histopathology Score') +
                              theme_bw(base_size = 20), 
                              NULL,
                            # hflf_toxin_plot +
                            #   labs(subtitle = 'Toxin',
                            #        x = 'Total Toxin (mean-centered)') +
                            #   theme_bw(base_size = 20) +
                            #   theme(plot.title = element_blank(),
                            #         axis.title.y = element_blank()),
                            nrow = 1,
                            labels = c('b)', ''),
                            label_size = 22,
                            align = c('hv'),
                            axis = c('tblr')))

(figS2 <- plot_grid(supp_fig2a,
                   figS2_bottom_row,
                   ncol = 1,
                   rel_heights = c(1, 0.5),
                   labels = c('a)', '', ''),
                   label_size = 20) +
         theme(plot.background = element_rect(fill = "white", color = "white")))
```

## **Supplemental Figure 3**

**manuscript specific edits to plots**
```{r}
uu_resil_stat_vis_v2 <- uu_resil_stat_vis +
                          scale_fill_gradient2(high = "#0D0887FF",
                                               low = "#CC4678FF",
                                               name = 'Group 1 -\nGroup 2',
                                               trans = 'reverse') +
                          theme(legend.position = "None")

uu_homog_stat_vis_v2 <- uu_homog_stat_vis +
                          scale_fill_gradient2(high = "#0D0887FF",
                                               low = "#CC4678FF",
                                               name = 'Group 1 -\nGroup 2',
                                               trans = 'reverse')

unweighted_pcoa_v2 <- unweighted_pcoa +
                        theme_bw(base_size = 20) +
                        theme(strip.text.y.right = element_text(angle = 0)) +
                        scale_fill_viridis(option = 'C', name = "Faith's\nPD")
```
**actual manuscript figure**
```{r, warning=FALSE, fig.width=28, fig.height=18}
## use NULL to create a ghost plot to alter your rows
figS3_bottom_row <- plot_grid(uu_resil_plot, uu_homog_plot,
                             uu_resil_stat_vis_v2, uu_homog_stat_vis_v2,
                             nrow = 2,
                             rel_heights = c(1, 0.7),
                             rel_widths = c(0.90, 1),
                             labels = c('d)', 'e)', 'f)', 'g)'),
                             label_size = 20)

figS3_top_right <- plot_grid(lacto_contam_plot,
                             lacto_stat_plot,
                             labels = c('a)', 'b)'),
                             label_size = 20,
                             ncol = 1,
                             rel_heights = c(1, 0.8))

                    
figS3_top_row <- plot_grid(figS3_top_right,
                          unweighted_pcoa_v2,
                          nrow = 1,
                          rel_widths = c(0.9, 1),
                          labels = c('', 'c)'),
                          label_size = 20)

plot_grid(figS3_top_row,
          figS3_bottom_row,
          nrow = 2,
          rel_heights = c(0.9, 1)) -> figS3

figS3
```

## **Supplemental Figure 6**

**manuscript specific edits to plots**
```{r}
newExp_dietVendor_surv_plot_v2 <- newExp_dietVendor_surv_plot +
                                    labs(title = 'Mouse Survival After Infection by Diet and Vendor',
                                         subtitle = 'AMC Follow-Up Experiments',
                                         x = 'Days Post Infection',
                                         y = 'Survival Probability')

newExp_avWeight_plot_v2 <- newExp_avWeight_plot +
                            labs(title = 'Weight Change After Infection by Diet',
                                 x = 'Days Post Infection')
```

**actual manuscript figure**
```{r, fig.height=17, fig.width=17}
(figS6 <- plot_grid(newExp_dietVendor_surv_plot_v2,
                    newExp_avWeight_plot_v2,
                    labels = c('a)', 'b)'),
                    label_size = 20,
                    nrow = 2,
                    rel_heights = c(0.70, 1),
                    align = 'hv',
                    axis = 'tblr'))
```

## **Supplemental Figure 7**
stats for fig6c (microbiome comparisons between all experiments done)
**Figure S7**
```{r, fig.height=20, fig.width=28}
mini_d3_genus_relAbun_stat_plot_v2 <- mini_d3_genus_relAbun_stat_plot +
                                          scale_fill_gradient2(low = "#0D0887FF",
                                                               high = "#CC4678FF",
                                                               name = 'Group 1 -\nGroup 2') +
                                          scale_y_discrete(labels = c('AMC-FU:CR',
                                                                      'AMC-FU:T',
                                                                      'UA:CR')) +
                                          scale_x_discrete(labels = c('AMC-\nP:T',
                                                                      'AMC-\nFU:CR',
                                                                      'AMC-\nFU:T')) +
                                          ggtitle('All Experiment Microbe Comparisons')

mini_d3_btDiets_relAbun_stat_plot_v2 <- mini_d3_btDiets_relAbun_stat_plot +
                                          scale_fill_gradient2(low = "#0D0887FF",
                                                               high = "#CC4678FF",
                                                               name = 'Group 1 -\nGroup 2')

## changing the size of the points in the boxplot so they're not as tiny
cdiff_genusAbun_plot$layers[[3]]$aes_params$size <- 3


figS7b <- ggdraw(add_sub(plot = mini_d3_genus_relAbun_stat_plot_v2,
                          label = "**CR: Charles River\n**T: Taconic",
                          x = 0, hjust = 0, fontface = "bold", size=18))

figS7_top_row <- plot_grid(cdiff_genusAbun_plot, NULL,
                           nrow = 1,
                           rel_widths = c(1, 0.1),
                           align = c('hv'),
                           axis = c('tblr'))

figS7_bottom_row <- plot_grid(figS7b,
                              mini_d3_btDiets_relAbun_stat_plot_v2,
                              nrow = 1,
                              rel_widths = c(0.65, 1),
                              labels = c('b)', 'c)'),
                              label_size = 22)

(figS7 <- plot_grid(figS7_top_row,
                    figS7_bottom_row,
                    nrow = 2,
                    labels = c('a)', ' '),
                    label_size = 22,
                    rel_heights = c(0.7, 1)) +
         theme(plot.background = element_rect(fill = "white", color = "white")))
```

## **Saving my Outputs**
as a .pdf and not an rdata object (just to clarify)
- had to also save plots as pngs bc word is stupid and wont render pdfs anymore 
```{r}
ggsave('fig1.pdf',
       plot = fig1,
       width = 15,
       height = 15)
ggsave('pngs/fig1.png',
       plot = fig1,
       width = 15,
       height = 15)

ggsave('fig2.pdf',
       plot = fig2,
       width = 20,
       height = 11)
ggsave('short_fig2.pdf',
       plot = fig2_top_row,
       width = 20,
       height = 6.5)
ggsave('pngs/fig2.png',
       plot = fig2,
       width = 20,
       height = 11)
ggsave('pngs/short_fig2.png',
       plot = fig2_top_row,
       width = 20,
       height = 6.5)

## new figure 3!
ggsave('fig3.pdf',
       plot = fig3,
       width = 24,
       height = 23)
ggsave('pngs/fig3.png',
       plot = fig3,
       width = 24,
       height = 23)

## new figure 5!
ggsave('fig4.pdf',
       plot = fig4,
       width = 15,
       height = 14)
ggsave('pngs/fig4.png',
       plot = fig4,
       width = 15,
       height = 14)

ggsave('fig5.pdf',
       plot = fig5,
       width = 20,
       height = 23)
ggsave('pngs/fig5.png',
       plot = fig5,
       width = 20,
       height = 23)

ggsave('fig6.pdf',
       plot = new_fig6,
       width = 27,
       height = 22)
ggsave('pngs/fig6.png',
       plot = new_fig6,
       width = 27,
       height = 22)

ggsave('fig7.pdf',
       plot = fig7,
       width = 15,
       height = 15)
ggsave('pngs/fig7.png',
       plot = fig7,
       width = 15,
       height = 15)

## used to be supplemental figures 3 and 4 but they got combined
ggsave('figS2.pdf',
       plot = figS2,
       width = 19,
       height = 17)
ggsave('pngs/figS2.png',
       plot = figS2,
       width = 19,
       height = 17)

## this used to be figure 4 but it got moved to supplemental
ggsave('figS3.pdf',
       plot = figS3,
       width = 10,
       height = 6)
 ggsave('pngs/figS3.png',
       plot = figS3,
       width = 10,
       height = 6)

## this used to be figure 6 but it got moved to supplemental 
ggsave('figS3.pdf',
       plot = figS3,
       width = 28,
       height = 18)
ggsave('pngs/figS3.png',
       plot = figS3,
       width = 28,
       height = 18)

ggsave('figS6.pdf',
       plot = figS6,
       width = 17,
       height = 17)
ggsave('pngs/figS6.png',
       plot = figS6,
       width = 17,
       height = 17)

ggsave('figS7.pdf',
       plot = figS7,
       width = 28,
       height = 20)
ggsave('pngs/figS7.png',
       plot = figS7,
       width = 28,
       height = 20)
```


## **Figure Graveyard**
past figures that have now been changed but I want to keep the code to generate them just in case

### **Old Fig6**
**manuscript specific edits to plots**
```{r, fig.width=15, fig.height=8}
my_viridis_c_palette <- c("#0D0887FF",
                          "#5D01A6FF",
                          "#9C179EFF",
                          "#CC4678FF",
                          "#ED7953FF",
                          "#FDB32FFF",
                          "#999999")

(everything_plot_v2 <- everything_plot +
                        ggtitle('AMC Follow-Up Experiment Culture and Survival Results') +
                        scale_color_manual(breaks = c('Enterococcus',
                                    'Escherichia-Shigella',
                                    'Proteus',
                                    'enterococcus_escherichiaShigella_proteus',
                                    'Lactobacillus',
                                    'Other',
                                    'None'),
                         labels = c('Enterococcus',
                                    'Escherichia-Shigella',
                                    'Proteus',
                                    'EC-ES-P',
                                    'Lactobacillus',
                                    'Other',
                                    'None'),
                         values = my_viridis_c_palette,
                         name = 'Genus') +
                        labs(tag = "**EC-ES-P: Enterococcus, Escherichia-Shigella, and Proteus") +
                        theme(plot.tag.position = c(0.64, 0.30),
                              plot.tag = element_text(size = 18, face = "bold")))

(everything_stats_v2 <- everything_stats +
                          ggtitle('AMC Follow-Up Experiment Culture Comparisons') +
                          scale_fill_gradient2(low = "#0D0887FF",
                                               high = "#CC4678FF",
                                               name = 'Group 1 -\nGroup 2'))
```

**actual manuscript figure**
```{r, fig.width=20, fig.height=17}
fig6_bottom_row <- plot_grid(everything_stats_v2, NULL,
                             nrow = 1,
                             rel_widths = c(1, 0.7))

(fig6 <- plot_grid(everything_plot_v2,
                  fig6_bottom_row,
                  ncol = 1,
                  labels = c('a)', 'b)'),
                  label_size = 20,
                  rel_heights = c(1, 0.5)) +
         theme(plot.background = element_rect(fill = "white", color = "white")))

ggsave('old_fig6.pdf',
       plot = fig6,
       width = 20,
       height = 17)
ggsave('pngs/old_fig6.png',
       plot = fig6,
       width = 20,
       height = 17)
```

### **Old Fig7**
**manuscript specific edits to plots**
```{r, fig.height=15, fig.width=15}
miniVendor_genusAbun_plot_v2 <- miniVendor_genusAbun_plot +
                                  scale_fill_manual(values = c("#DB5C68FF", "#FEBC2AFF"),
                                                    name = 'Mouse\nVendor',
                                                    labels = c('Charles\nRiver', 'Taconic')) +
                                  scale_x_continuous(breaks = c(1, 2, 3),
                                                     labels = c('AMC-P',
                                                                'UA',
                                                                'AMC-FU')) +
                                  ggtitle('All Experiment Microbes by Vendor')

(miniVendor_genusAbun_plot_v2_w_legend <- ggdraw(add_sub(plot = miniVendor_genusAbun_plot_v2,
               label = "**AMC-P: Previous AMC experiments (published)\n**UA: University of Arizona experiments\n**AMC-FU: AMC follow-up experiments",
               x = 0, hjust = 0, fontface = "bold", size=18)))


mini_d3_genus_relAbun_stat_plot_v2 <- mini_d3_genus_relAbun_stat_plot +
                                          scale_fill_gradient2(low = "#0D0887FF",
                                                               high = "#CC4678FF",
                                                               name = 'Group 1 -\nGroup 2') +
                                          scale_y_discrete(labels = c('AMC-FU:CR',
                                                                      'AMC-FU:T',
                                                                      'UA:CR')) +
                                          scale_x_discrete(labels = c('AMC-\nP:T',
                                                                      'AMC-\nFU:CR',
                                                                      'AMC-\nFU:T')) +
                                          ggtitle('All Experiment Microbe Comparisons')

(mini_d3_genus_relAbun_stat_plot_v2_w_legend <- ggdraw(add_sub(plot = mini_d3_genus_relAbun_stat_plot_v2,
                          label = "**CR: Charles River\n**T: Taconic",
                          x = 0, hjust = 0, fontface = "bold", size=18)))
```

Comparison analysis between first set anschutz, u of arizona, and new anschutz experiment microbiome data at day -15 and day 3 with stats.

**actual manuscript figure**
```{r, fig.height=13, fig.width=25}
(fig7 <- plot_grid(miniVendor_genusAbun_plot_v2_w_legend, 
                  mini_d3_genus_relAbun_stat_plot_v2_w_legend,
                  labels = c('a)', 'b)'),
                  label_size = 20,
                  nrow = 1,
                  rel_heights = c(1, 1)))
```

### **Old FigS3**
hypoxia data that was taken out of the paper
```{r, fig.height=6, fig.width=10}
figS3 <- plot_grid(hypoxia_cecum_plot +
                        theme_bw(base_size = 20) +
                        ggtitle('Cecum Hypoxia Between Diets'), 
                      NULL,
                      labels = c('a)', ' '),
                      axis = c('tblr'),
                      align = c('hv'),
                      label_size = 20,
                      rel_widths = c(1, 0.5)) +
        theme(plot.background = element_rect(fill = "white", color = "white"))

figS3
```