---
title: "manuscript_figures"
output: html_document
date: "2023-10-02"
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

```{r}
library(tidyverse)
library(ggplot2)
library(cowplot)
library(apppleplots)
```

# **Putting Together Figures for Paper Manuscript**


## **Figure 1**
Plots to go in figure:
- experimental timeline diagram 
- mouse survival curve plot and stats

*Caption*:
*a) Survival curves for the five tested diets after antibiotic induced gut microbiome dysbiosis and C. difficile infection. b) A heat map of log-normal pairwise comparisons of survival probability between the tested diets. P-value significance (****p<0.0001, ***p<0.001, **p<0.01, *p<0.05, ns<1).*

```{r, warning=FALSE, fig.height=12, fig.width=15}
fig1_bottom_row <- plot_grid(surv_stat_vis, NULL,
                             rel_widths = c(0.6, 1),
                             labels = c('b)', ' '),
                             label_size = 20)

plot_grid(diet_plot_final +
            theme_bw(base_size = 18),
          fig1_bottom_row,
          rel_heights = c(1, 0.5),
          nrow = 2,
          labels = c('a)', ' '),
          align = 'h',
          label_size = 20) -> fig1

fig1
```

## **Figure 2**
Plots to go in figure:
- cecal toxin plot 
- cecal overall histopathology plot
- cecal toxin/histo correlation plot (p-value significance should be overlaid) - TAKE OUT!!

*Caption*:
*a) C. difficile TcdA and TcdB toxin concentrations in the five tested diets, determined by ELISA. b) Histologic inflammation scoring of mouse cecal and colon contents at three days post C. difficile infection in the tested diets as determined by a blinded histologist. c) Linear regression comparison of C. difficile toxins TcdA/TcdB concentration to histologic inflammation scores. Statistical analysis was done using a Kruskal-Wallis test followed by a Dunn's Post Hoc test with p-values being corrected for multiple comparisons via Benjamini and Hochberg. P-value significance* (****p<0.0001, ***p<0.001, **p<0.01, *p<0.05, ns<1).*
changing the size of the labels on the plot
```{r}
noDiet_histoCecum_toxStats_plot$layers[[4]]$aes_params$size <- 5.5
```

```{r, warning=FALSE, fig.height=11, fig.width=20}
fig2_top_row <- plot_grid(neat_plot +
                            theme_bw(base_size = 20) +
                            ylab('Concentration (ng/uL)') +
                            ggtitle('Cecum C. difficile Toxin Concentration'),
                          all_day_plot +
                            theme_bw(base_size = 20) +
                            ggtitle('Histopathology Scores') +
                            ylab('Score'),
                             labels = c('a)', 'b)'),
                             align = c('hv'),
                             axis = c('tblr'),
                             nrow = 1,
                            rel_widths = c(0.9, 1),
                             label_size = 20)

fig2_bottom_row <- plot_grid(noDiet_histoCecum_toxStats_plot +
                               theme_bw(base_size = 20),
                            labels = c('c)'),
                            label_size = 20)
plot_grid(fig2_top_row,
          fig2_bottom_row,
          ncol = 1,
          align = c('hv'),
          axis = c('tblr'),
          rel_heights = c(1, 0.7)) -> fig2

fig2
```

## **Figure 3**
Plots to go in figure:
- bile acid concentration plot
- bile acid c. diff promoters/inhibitors ratio plot
- bile acid/toxin correlation comparison plots 
- metabolomics plot

*Caption*:
*a) Bile acid cecal levels of C. difficile growth inhibitors (a-MCA, b-MCA, LCA, DCA) and C. difficile growth promoters (TCA, CA) across the tested diets. b) The ratios of C. difficile growth promoters:inhibitors across the tested diets (a higher ratio indicates a higher level of promoters relative to inhibitors for that diet). Statistical analysis for dietary bile acid concentration/ratio differences was done using a Kruskal-Wallis test followed by a Dunn's Post Hoc test with p-values being corrected for multiple comparisons via Benjamini and Hochberg. Linear regression comparison of C. difficile toxins TcdA/TcdB concentration to c) C. difficile growth inhibitors and d) C. difficile growth promoters. e) SCFA (acetate, butyrate, propionate) levels of mouse cecal contents three days post C. difficile infecion by the tested diets. P-value significance* (****p<0.0001, ***p<0.001, **p<0.01, *p<0.05, ns<1)*

```{r, fig.height=17, fig.width=20}
fig3_top_row <- plot_grid(dca_sep_plot +
                            theme_bw(base_size = 20), 
                          ratio_bile_plot +
                            theme_bw(base_size = 20)+
                            ggtitle('Cecum Bile Acid Ratio'),
                          labels = c('a)', 'b)'),
                          label_size = 20,
                          align = c('hv'),
                          axis = c('tblr'),
                          rel_widths = c(1, 0.5))

fig3_middle_row <- plot_grid(panelC_fig3_redoWithlab,
                             ncol = 1,
                             labels = c('c)'),
                             label_size = 20,
                             align = c('hv'),
                             axis = c('tblr'))


fig3_bottom_row <- plot_grid(butyrateOnly_plot +
                               theme_bw(base_size = 20) +
                               ggtitle('Cecum Butyrate Concentration'),
                             panelE_fig3_redoWithlab,
                             rel_widths = c(0.4, 1),
                             nrow = 1,
                             labels = c('d)', 'e)'),
                             label_size = 20)

fig3 <- plot_grid(fig3_top_row,
                  fig3_middle_row,
                  fig3_bottom_row,
                  ncol = 1,
                  rel_heights = c(0.9, 0.8, 0.9),
                  align = c('hv'),
                  axis = c('tblr'))

fig3
```

## **Figure 4**
NOTE: stats have been redone log10 transformed

Plots to go in figure:
- faith's pd and stats

how you change the size of the labels in a ggplot object!!
```{r}
faith_combStats_plot$layers[[4]]$aes_params$size <- 5
```


*Caption*:
*a) Faith's Phylogenetic Diversity (alpha diversity) analysis by the five tested diets. The dashed red line at day negative three indicates the day the mice received oral antibiotics and the purple dashed line at day zero indicates with the mice were infected with C. difficile. b) Pairwise comparisons between diets as determined by a Kruskal-Wallis test followed by a Dunn's Post Hoc test with the color gradient indicating the difference in means.*P-value significance* (****p<0.0001, ***p<0.001, **p<0.01, *p<0.05, ns<1).*
```{r, warning=FALSE, fig.height=14, fig.width=15}
plot_grid(faith_plot +
            theme_bw(base_size = 20), 
          faith_stat_vis,
          faith_combStats_plot,
          ncol = 1,
          rel_heights = c(1, 0.7, 1),
          labels = c('a)', 'b)', 'c)'),
          align = 'h',
          label_size = 20) -> fig4

fig4
```

## **Figure 5**
NOTE: stats have already been redone log10 transformed!!

Plots to go in figure:
- want to eventually have the plot that goes with 16S for stool and cecal samples 
- need an unbiased differential abundance test? 
- asv level of what things are influenced by diets 
- look for microbes that correlate with inflammation - use an ANCOMBC 
- filter low abundance asvs (anything that's in less than 20% of the samples)
**- have a distance matrix for every single mouse by their taxonomic information via qiime or r (which taxa are showing more related patterns by diet in mice). then, run an adonis test on it.**

*Caption*:
*a) Relative abundances of key gut bacterial families by the five tested diets throughout the experimental timeline. The dashed red line at day negative three indicates the day the mice received oral antibiotics and the purple dashed line at day zero indicates with the mice were infected with C. difficile. b) Pairwise comparisons by days relative to infection (columns) and bacterial family (rows) per diet as determined by a Kruskal-Wallis test followed by a Dunn's Post Hoc test with the color gradient indicating the difference in means.*P-value significance* (****p<0.0001, ***p<0.001, **p<0.01, *p<0.05, ns<1).*
```{r}
microbeProd_relAbunStats_plot$layers[[6]]$aes_params$size <- 5
```

```{r, warning=FALSE, fig.height=30, fig.width=29}
plot_grid(genusAbun_plots_together_withLabs,
          microbeProd_relAbunStats_plot,
          labels = c(' ', 'c)'),
          ncol = 1,
          label_size = 20,
          rel_heights = c(1, 0.4)) -> fig5

fig5
```

```{r, fig.height=30, fig.width=29}
fig5_part_two <- plot_grid(lachno_abun_plot +
                            theme(axis.title.x = element_blank()) +
                             labs(y = '',
                                  subtitle = 'Lachnospiraceae',
                                  title = 'Obligate Anaerobe Relative Abundance'),
                            rumino_abun_plot +
                              theme(plot.title = element_blank()) +
                              labs(x = 'Days Relative to Infection',
                                   y = '',
                                   subtitle = 'Ruminococcaceae'),
                            path_abun_plot,
                            microbeProd_relAbunStats_plot,
                            ncol = 1,
                            labels = c('a)', ' ', 'b)', 'c)'),
                            label_size = 20,
                            rel_heights = c(0.5, 0.5, 1, 0.8),
                            align = 'hv',
                            axis = 'tblr')

fig5_part_two_withLabs <- ggdraw(fig5_part_two) +
  draw_label('Relative Abundace (log10)', x = 0, y = 0.82, angle = 90, size = 20, vjust = 1.5)

fig5_part_two_withLabs
```


## **Figure 6**
New experiment culture/mortality results.
```{r, fig.width=22, fig.height=17}
fig6_bottom_row <- plot_grid(everything_stats, NULL,
                             nrow = 1,
                             rel_widths = c(1, 0.7))

fig6 <- plot_grid(everything_plot,
                  fig6_bottom_row,
                  ncol = 1,
                  labels = c('a)', 'b)'),
                  label_size = 20,
                  rel_heights = c(1, 0.5))
fig6
```

```{r, fig.width=30, fig.height=12}
test_fig6_bottom_row <- plot_grid(everything_stats, NULL,
                             ncol = 1,
                             rel_heights = c(0.7, 1))

test_fig6 <- plot_grid(everything_plot,
                  test_fig6_bottom_row,
                  nrow = 1,
                  labels = c('a)', 'b)'),
                  label_size = 20,
                  rel_widths = c(1, 0.4))

test_fig6
```

## **Figure 7**
Comparison analysis between first set anschutz, u of arizona, and new anschutz experiment microbiome data at day -15 and day 3 with stats.
```{r, fig.height=13, fig.width=30}
fig7 <- plot_grid(miniVendor_genusAbun_plot +
                    ggtitle('All Experiment Microbes by Vendor'), 
                  mini_d3_genus_relAbun_stat_plot +
                    scale_y_discrete(labels = c('New Anschutz\n(2024) - Charles River',
                                                'New Anschutz\n(2024) - Taconic',
                                                'U of Arizona\n- Charles River')) +
                    ggtitle('All Experiment Microbe Comparisons'),
                  labels = c('a)', 'b)'),
                  label_size = 20,
                  nrow = 1,
                  # align = 'hv',
                  # axis = 'tblr',
                  rel_heights = c(1, 1))

fig7
```
this just isnt working ugh idk 
```{r, fig.width=30, fig.height=25}
plot_grid(test_fig6,
          fig7,
          nrow = 2,
          align = 'hv',
          axis = 'tblr')
```

## **Supplemental Figure 5**
Plots to go in figure:
- overall hypoxia plot (location on x-axis)?
- cecum hypoxia plot (diet on x-axis)
- cecal hypoxia/histopathology correlation comparison plot (stats as supplementary since nothing is significant?)
- only include panels b and e going forward!!

*Caption*:
*a) Hypoxia measurements for three intestinal locations (cecum, proximal colon, distal colon) in the tested diets, determined by fluorescence of the oxygen sensitive dye, pimonidazole, in the respective tissues. b) Cecal hypoxia measurements between the five tested diets. Statistical analysis was done using a Kruskal-Wallis test followed by a Dunn's Post Hoc test with p-values being corrected for multiple comparisons via Benjamini and Hochberg. c) Linear regression comparison of cecal histologic inflammation scoring to hypoxia across tested diets. d) Linear regression comparison o colon histologic inflammation scoring to hypoxia across tested diets. P-value significance* (****p<0.0001, ***p<0.001, **p<0.01, *p<0.05, ns<1).

**Figure S5**
```{r, fig.height=6, fig.width=10}
figS5 <- plot_grid(hypoxia_cecum_plot +
                        theme_bw(base_size = 20) +
                        ggtitle('Cecum Hypoxia Between Diets'), 
                      NULL,
                      labels = c('a)', ' '),
                      axis = c('tblr'),
                      align = c('hv'),
                      label_size = 20,
                      rel_widths = c(1, 0.5))

figS5
```


## **Supplemental Figure 7**

*Caption*:
*a) Unweighted UniFrac PCoA plots delimited by diet (columns) and by days relative to infection (rows). Individual points are colored by their associated Faith's PD score. b) Microbiome resilience determined using pairwise comparisons between day negative eight (seven days post diet switch and five days prior to oral antibiotic administration) and later time points. c) Microbiome homogeneity determined using pairwise Unweighted UniFrac distances within each time point by tested diets. The dashed red line at day negative three indicates the day the mice received oral antibiotics and the purple dashed line at day zero indicates with the mice were infected with C. difficile. Statistical analysis of microbiome d) resiliency and e) homogeneity was done with a Kruskal-Wallis test followed by a Dunn's Post Hoc test with the color gradient indicating the difference in means between the various groups.*P-value significance* (****p<0.0001, ***p<0.001, **p<0.01, *p<0.05, ns<1).*
```{r, warning=FALSE, fig.width=26.5, fig.height=18}
## use NULL to create a ghost plot to alter your rows
figS7_bottom_row <- plot_grid(uu_resil_plot, uu_homog_plot,
                             uu_resil_stat_vis +
                               theme(legend.position = "None"), 
                             uu_homog_stat_vis,
                             nrow = 2,
                             rel_heights = c(1, 0.7),
                             rel_widths = c(0.90, 1),
                             labels = c('b)', 'c)', 'd)', 'e)'),
                             label_size = 20)

                    
figS7_top_row <- plot_grid(unweighted_pcoa,
                          NULL,
                          nrow = 1,
                          rel_widths = c(1, 0.5),
                          labels = c('a)', ' '),
                          label_size = 20,
                          align = c('hv'),
                          axis = c('tblr'))

plot_grid(figS7_top_row,
          figS7_bottom_row,
          nrow = 2,
          rel_heights = c(0.9, 1)) -> figS7

figS7
```

## **Supplemental Figure 10**

*Caption:*
*Mouse survival between diets for mice from the two different vendors. We conducted another round of mouse experiments in 2024 at the Anschutz Medical Campus to investigate the differences in survival results seen between our previous experiment and this one. This plot shows how mouse survival was not significantly different between vendors (except for in the LF/HF diet which had a p-value of 0.03) and that the survival trends are similar to what is depicted in Figures 1 and S7. A Cox Proportional Hazards (PH) regression model indicated that only mice on the low-fat/low-fiber diet (hazard ratio = 3.66, p-value = 0.004) had significantly worse survival as compared to mice on chow. P-value significance determined via Kaplan-Meier tests, shaded area shows a 95% CI (**** : p<0.0001, *** : p<0.001, ** : p<0.01, * : p<0.05, ns : p≥0.05).*

```{r, fig.height=17, fig.width=17}
(figS10 <- plot_grid(newExp_dietVendor_surv_plot,
          newExp_avWeight_plot +
            labs(title = 'Weight Change After Infection by Diet',
                 x = 'Days Post Infection'),
          labels = c('a)', 'b)'),
          label_size = 20,
          nrow = 2,
          rel_heights = c(0.70, 1),
          align = 'hv',
          axis = 'tblr'))
```



**Saving my Outputs**
as a .pdf and not an rdata object (just to clarify)
```{r}
ggsave('fig1.pdf',
       plot = fig1,
       width = 15,
       height = 12)

ggsave('fig2.pdf',
       plot = fig2,
       width = 20,
       height = 11)
ggsave('short_fig2.pdf',
       plot = fig2_top_row,
       width = 20,
       height = 6.5)

## new figure 3!
ggsave('fig3.pdf',
       plot = fig3,
       width = 20,
       height = 17)

## new figure 5!
ggsave('fig4.pdf',
       plot = fig4,
       width = 15,
       height = 14)

ggsave('fig5.pdf',
       plot = fig5,
       width = 29,
       height = 30)

ggsave('fig6.pdf',
       plot = fig6,
       width = 22,
       height = 17)

ggsave('fig7.pdf',
       plot = fig7,
       width = 30,
       height = 13)

## this used to be figure 4 but it got moved to supplemental
ggsave('figS5.pdf',
       plot = figS5,
       width = 10,
       height = 6)

## this used to be figure 6 but it got moved to supplemental 
ggsave('figS7.pdf',
       plot = figS7,
       width = 26.5,
       height = 18)

ggsave('figS10.pdf',
       plot = figS10,
       width = 17,
       height = 17)
```

